pizza_cutter_metadetect/masks.py

```diff
@@ -0,0 +1,450 @@
+    )
+
+    # set the right entries in the global mask image
+    msk_img[
+        slice_bnds["min_row"]+srow:slice_bnds["max_row"]+srow,
+        slice_bnds["min_col"]+scol:slice_bnds["max_col"]+scol,
+    ] |= slice_msk[
+        slice_bnds["min_row"]:slice_bnds["max_row"],
+        slice_bnds["min_col"]:slice_bnds["max_col"],
+    ]
+
+
+def _mask_one_slice(
+    *,
+    buffer_size,
+    central_size,
+    coadd_dims,
+    msk_img,
+    scol,
+    srow,
+):
+    slice_bnds = get_slice_bounds(
+        orig_start_col=scol,
+        orig_start_row=srow,
+        central_size=central_size,
+        buffer_size=buffer_size,
+        coadd_dims=coadd_dims,
+    )
+    msk_img[
+        slice_bnds["min_row"]+srow:slice_bnds["max_row"]+srow,
+        slice_bnds["min_col"]+scol:slice_bnds["max_col"]+scol,
+    ] |= MASK_NOSLICE
+
+
+def make_mask(
+    *,
+    preconfig,
+    missing_slice_inds,
+    obj_data,
+    central_size,
+    buffer_size,
+    wcs,
+    position_offset,
+    coadd_dims,
+    info,
+    gaia_stars=None,
+    healpix_nside=131072,
+):
+    """Make the healsparse mask indicating where measurements were made.
+
+    Parameters
+    ----------
+    preconfig : dict
+        Proprocessing configuration. Must contain gaia_star_masks
+        entry if gaia_stars is not None.
+    missing_slice_inds : list of int
+        The indices into `obj_data` indicating slices that did not get measurements.
+    obj_data : structure numpy array
+        The object data from the MEDS file. Used for getting the slice locations
+        orig_start_col/orig_start_row. These values are the same across bands and
+        so the data from any of the bands is fine.
+    central_size : int
+        The size of the central region of the slice.
+    buffer_size : int
+        The size of the buffer around each slice.
+    wcs : esutil.wcsutil.WCS
+        A WCS object for the coadd tile.
+    position_offset : int
+        The position offset to convert from 0-indexed pixel coordinates into
+        the convention expected by the `wcs`.
+    coadd_dims : 2-tuple of ints
+        The dimension of the coadd image.
+    info : dict
+        The coadd tile info produced by the pizze cutter and stored in the MEDS
+        metadata as the 'tile_info' entry. This dict is used to get the coadd
+        tile unique ra/dec bounds (uramin, uramax, udecmin, udecmax) and if the
+        tile cross ra = 0 (crossra0).
+    gaia_stars : structured numpy array, optional
+        The GAIA star data indicating where the GAIA star mask holes are. The default
+        of None will result in no GAIA star mask holes.
+    healpix_nside : int, optional
+        The nisde of the healsparse mask. The default is 131072.
+
+    Returns
+    -------
+    hs_msk : healsparse map
+        The healsparse boolean map indicating what areas were masked and why. The
+        possible flag values are
+
+            MASK_NOSLICE = 2**0 - indicates no data for a given coadd slice
```

Comment:
These need to be updated, they are out of date.

Reply:
done